webapp/src/Service/DOMJudgeService.php

```diff
@@ -299,6 +299,27 @@ public function clearCookie(
         return $response;
     }
 
+    public function getUnreadClarifications(): array
+    {
+        $user           = $this->getUser();
+        $team           = $user->getTeam();
+        $clarifications = $team->getUnreadClarifications();
+        $contest        = $this->getCurrentContest($team->getTeamId());
+        $unreadClarifications = [];
+        if ($contest === null) {
+            return $unreadClarifications;
+        }
+        foreach ($clarifications as $clar) {
+            if ($clar->getContest() === $contest) {
```

Comment:
Did you check if this works? We might need
```suggestion
            if ($clar->getContest()->getCid() === $contest->getCid()) {
```

Reply:
I've checked it really works. But I think it needs to be fixed